### PR TITLE
Install .NET Core 3.1 SDK on Windows to support Track 1 SDK

### PR DESCRIPTION
**Pull request by live1206**
.NET Core 3.1 SDK does not exist on Windows hosted machine anymore, we still need it for Track 1 SDK
https://github.com/Azure/azure-sdk-for-net/pull/35148/checks?check_run_id=12290486264

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by live1206** [comment]
@weshaggard Could you help to review this? Thanks!

**Comment by jsquire** [comment]
@live1206:  Please help me understand why we need a runtime that has reached end-of-life and is no longer supported.   What libraries are failing and what target are they using?


//fyi: @heaths

**Review by jsquire — DISMISSED** [review]
Blocking merge until the scenario is understood.

**Comment by benbp** [comment]
If this is indeed not necessary, I have a PR to remove the step entirely as we don't need to be wasting time on windows jobs installing it: https://github.com/Azure/azure-sdk-for-net/pull/35174

**Comment by live1206** [comment]
> @live1206 I thought we were rolling forward to the latest .NET when running the code generation so why would we still need this? Is there something else that needs it?
> 
> Looking at your link it appears as though there are still some test projects targeting .NET Core 3.1. We should instead update those test projects to .NET 7.0 as .NET 3.1 is out support.

Tried with a draft PR https://github.com/Azure/azure-sdk-for-net/pull/35178, it takes more effort than expected. 
Basically a follow-up of https://github.com/Azure/azure-sdk-for-net/pull/32814

**Comment by jsquire** [comment]
> Tried with a draft PR #35178, it takes more effort than expected. Basically a follow-up of #32814

There's a difference between building with the .NET 7 SDK and changing the target frameworks for libraries.   We need to do the former, we do not want to do the latter.   @live1206 - would you please help us understand what problem you're trying to solve?  What behavior are you seeing that requires changes?

**Comment by live1206** [comment]
> https://github.com/Azure/azure-sdk-for-net/pull/35148/checks?check_run_id=12290486264

@jsquire As I mentioned in the description, the CI is failing for Track 1 .NET SDK PR: https://github.com/Azure/azure-sdk-for-net/pull/35148/checks?check_run_id=12290486264, it is blocking Track 1 .NET SDK release.

**Comment by jsquire** [comment]
> @jsquire As I mentioned in the description, the CI is failing for Track 1 .NET SDK

What should change here is the SDK in use; these should be compiled using the .NET 7 SDK.   We do not want to install/use the .NET Core 3.1 SDK, as it has reached end-of-life.

**Comment by live1206** [comment]
> > @jsquire As I mentioned in the description, the CI is failing for Track 1 .NET SDK
> 
> What should change here is the SDK in use; these should be compiled using the .NET 7 SDK. We do not want to install/use the .NET Core 3.1 SDK, as it has reached end-of-life.

As I explained in https://github.com/Azure/azure-sdk-for-net/pull/35178#discussion_r1151655504, it's not an easy fix to upgrade the target .NET to latest.

File: eng/pipelines/templates/steps/install-dotnet.yml
Comment (weshaggard) [inline eng/pipelines/templates/steps/install-dotnet.yml:30]: To help unblock the current release let's put the windows install of this 3.1 directly in the ci.mgmt.yml file to scope it to only track 1 mgmt. Once that is done we can continue to remove this entire step from install-dotnet.yml, happening in https://github.com/Azure/azure-sdk-for-net/pull/35151. Once that happens we can keep the 3.1 install for all OS's in the ci.mgmt.yml file temporarily. We should continue to work on https://github.com/Azure/azure-sdk-for-net/pull/35178 to break this dependency on 3.1 before we can no longer use this EoL framemwork.

Comment (live1206) [reply eng/pipelines/templates/steps/install-dotnet.yml]: Added a step in .NET SDK Track 1 CI to install .NET core 3.1 for all OS, so that we can continue with #35174 for the rest.

File: eng/pipelines/templates/jobs/ci.mgmt.yml
Comment (weshaggard) [inline eng/pipelines/templates/jobs/ci.mgmt.yml:19]: condition succeeded() is redundant but doesn't hurt anything either.

**Review by weshaggard — APPROVED** [review]

File: eng/pipelines/templates/jobs/ci.mgmt.yml
Comment (weshaggard) [inline eng/pipelines/templates/jobs/ci.mgmt.yml:18]: I suspect you likely need this in the Test job below as well.

**Comment by weshaggard** [comment]
I'd like to make sure we get a test run of the mgmt pipeline as well to ensure this change works as expected.

File: eng/pipelines/templates/jobs/ci.mgmt.yml
Comment (weshaggard) [reply eng/pipelines/templates/jobs/ci.mgmt.yml:19]: Looks like this condition does need to be removed because the and operator requires 2 arguments https://dev.azure.com/azure-sdk/public/_build/results?buildId=2677828&view=results

Comment (live1206) [reply eng/pipelines/templates/jobs/ci.mgmt.yml]: Good catch, removed.

File: eng/pipelines/templates/jobs/ci.mgmt.yml
Comment (live1206) [reply eng/pipelines/templates/jobs/ci.mgmt.yml:18]: Yes, it is needed, wondering why the test run did not fail though.

Comment (weshaggard) [reply eng/pipelines/templates/jobs/ci.mgmt.yml:18]: CI failed to trigger because of the bad and() expression.

**Review by weshaggard — APPROVED** [review]

**Pull request merged**
